src/uct/ib/ud/base/ud_ep.h

```diff
@@ -228,10 +228,10 @@ struct uct_ud_ep {
          UCS_STATS_NODE_DECLARE(stats);
          UCT_UD_EP_HOOK_DECLARE(tx_hook);
     } tx;
-    struct {
+    struct UCS_S_PACKED{
```

Comment:
style: space b4 {

Reply:
i would also move "rx" and "ca" before "resend", since "rx" is fast path and we want to keep it aligned

Reply:
another option: move the fields of "ca" to upper level, put them near "flags" and "path_bits", to close the holds without using UCS_S_PACKED

Reply:
I don't think the second option will work, It's not possible to close 4 bytes padding inside resend without using UCS_S_PACKED:
```
(gdb) p sizeof(uct_ud_ep_t::resend)
$2 = 16
```

Reply:
right, i missed the conn_id, so never mind. we'll deal with ud_ep size optimization later